### PR TITLE
[CELEBORN-1542] Master supports to check the worker host pattern on worker registration

### PR DESCRIPTION
**Pull request by turboFei**
### What changes were proposed in this pull request?
This pr introduce an optional config item for worker host pattern, and support to check whether the worker host matches the pattern in master end when registering the worker.

If it does not match, the register worker request will be rejected.

### Why are the changes needed?
Currently, the celeborn master allow all the workers to register. It is better to limit the workers allowed to register. 

### Does this PR introduce _any_ user-facing change?

No, the config item is optional, no broken change.

### How was this patch tested?
UT.

### CODE REVIEW

**Comment by turboFei** [comment]
also cc @cfmcgrady

File: common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala [hunk 1]
```diff
@@ -2175,6 +2177,14 @@ object CelebornConf extends Logging {
       .timeConf(TimeUnit.MILLISECONDS)
       .createWithDefaultString("120s")
 
+  val WORKER_HOST_PATTERN: OptionalConfigEntry[Regex] =
+    buildConf("celeborn.master.workerHost.pattern")
```
Comment (pan3793) [inline common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2181]: how about?
```
celeborn.master.allowWorkerHostPattern
celeborn.master.denyWorkerHostPattern
```

Comment (turboFei) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala:2181]: will enhance it

Comment (turboFei) [reply common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala]: addressed

**Review by cfmcgrady — APPROVED** [review]

**Review by pan3793 — APPROVED** [review]

**Review by RexXiong — APPROVED** [review]

**Comment by RexXiong** [comment]
Thanks. merge to main(v0.6.0)

**Pull request closed without merging**